public/scripts/welcome-screen.js

```diff
@@ -0,0 +1,412 @@
+        return defaultAssistantAvatar;
+    }
+
+    return assistantAvatar;
+}
+
+export async function openWelcomeScreen() {
+    const currentChatId = getCurrentChatId();
+    if (currentChatId !== undefined || chat.length > 0) {
+        return;
+    }
+
+    await sendWelcomePanel();
+    sendAssistantMessage();
+    sendSystemMessage(system_message_types.WELCOME_PROMPT);
+}
+
+function sendAssistantMessage() {
+    const currentAssistantAvatar = getPermanentAssistantAvatar();
+    const character = characters.find(x => x.avatar === currentAssistantAvatar);
+    const name = character ? character.name : neutralCharacterName;
+    const avatar = character ? getThumbnailUrl('avatar', character.avatar) : system_avatar;
+
+    const message = {
+        name: name,
+        force_avatar: avatar,
+        mes: t`If you're connected to an API, try asking me something!` + '\n***\n' + t`**Hint:** Set any character as your welcome page assistant from their "More..." menu.`,
+        is_system: false,
+        is_user: false,
+        extra: {},
+    };
+
+    chat.push(message);
+    addOneMessage(message);
+}
+
+async function sendWelcomePanel() {
+    try {
+        const chatElement = document.getElementById('chat');
+        const sendTextArea = document.getElementById('send_textarea');
+        if (!chatElement) {
+            console.error('Chat element not found');
+            return;
+        }
+        const chats = await getRecentChats();
+        const templateData = {
+            chats,
+            empty: !chats.length,
+            version: displayVersion,
+            more: chats.some(chat => chat.hidden),
+        };
+        const template = await renderTemplateAsync('welcomePanel', templateData);
+        const fragment = document.createRange().createContextualFragment(template);
+        fragment.querySelectorAll('.welcomePanel').forEach((root) => {
+            const recentHiddenClass = 'recentHidden';
+            const recentHiddenKey = 'WelcomePage_RecentChatsHidden';
+            if (accountStorage.getItem(recentHiddenKey) === 'true') {
+                root.classList.add(recentHiddenClass);
+            }
+            root.querySelectorAll('.showRecentChats').forEach((button) => {
+                button.addEventListener('click', () => {
+                    root.classList.remove(recentHiddenClass);
+                    accountStorage.setItem(recentHiddenKey, 'false');
+                });
+            });
+            root.querySelectorAll('.hideRecentChats').forEach((button) => {
+                button.addEventListener('click', () => {
+                    root.classList.add(recentHiddenClass);
+                    accountStorage.setItem(recentHiddenKey, 'true');
+                });
+            });
+        });
+        fragment.querySelectorAll('.recentChat').forEach((item) => {
+            item.addEventListener('click', () => {
+                const avatarId = item.getAttribute('data-avatar');
```

Comment:
```suggestion
                const avatarId = item.dataset.avatar;
```
Doesn't this work? I mean, not that it matters too much. Usually find that cleaner though.

Reply:
querySelector returns Element type which has no dataset property.